src/topp/IDFileConverter.cpp

```diff
@@ -390,6 +397,37 @@ class TOPPIDFileConverter :
           }
         }
       }
+      else if (in_type == FileTypes::PSMS) // Percolator
+      {
+        String mz_file = getStringOption_("mz_file");
+        MSExperiment<> experiment;
+        MSExperiment<>* experiment_p = 0;
+        if (!mz_file.empty())
+        {
+          fh.loadExperiment(mz_file, experiment);
+          experiment_p = &experiment;
+        }
+        String score_type = getStringOption_("score_type");
+        enum PercolatorOutfile::ScoreType perc_score;
```

Comment:
instead of this if() blocks, you could add the string representations of the score types to the PercolatorOutFile.
This would avoid decoupling -- similar to http://ftp.mi.fu-berlin.de/pub/OpenMS/documentation/html/MassTrace_8h_source.html
and then
```
defaults_.setValue("quant_method", String(MassTrace::names_of_quantmethod[0]), "...");
defaults_.setValidStrings("quant_method", std::vector<String>(MassTrace::names_of_quantmethod, MassTrace::names_of_quantmethod +(int)MassTrace::SIZE_OF_MT_QUANTMETHOD));
```
to set the parameters as a one-liner